src/clusters.tsx

```diff
@@ -45,6 +46,9 @@ export class DaskClusterManager extends Widget {
   /**
    * Create a new Dask cluster manager.
    */
+  status = 'ready';
+  commands = new CommandRegistry()
```

Comment:
Instead of defining a new one, rewrite https://github.com/dask/dask-labextension/blob/master/src/index.ts#L373-L378

Reply:
The command registry is intended to be a application singleton, so we need to either
1. Pass in the command registry as a constructor option to the cluster manager for use in the toolbar (as I think you are suggesting here), or
2. Abandon the idea of a `CommandToolbarButton`, and use the lower-level approach like [here](https://github.com/jupyterlab/jupyterlab/blob/8289dd8b18a04db3889fb69287b617b2677b6338/packages/notebook/src/default-toolbar.tsx#L66-L87).

It's unfortunate that the regular `ToolbarButton` widget can't update its `enabled` state, I think it's a defect in its design. So until we fix that, I think we need one of the two above workarounds.

Reply:
I have already gone through some contortions to avoid the first option (having to pass in the `CommandRegistry`), see, e.g., [here](https://github.com/dask/dask-labextension/blob/4342a1bf088a8b20aff23c30d4e0d52eb6e10641/src/index.ts#L147-L149)

Reply:
Oops I missed this comment. I don't understand why 1 is undesirable.

Reply:
Well, it's not exactly a problem to pass in the command registry, so much as an effort to separate concerns. My thinking has been this:

The cluster manager should only need to know about things related to the management of dask clusters. If it has a handle to the command registry, then it can access the whole universe of commands that JupyterLab has, and it becomes more difficult to reason about what the cluster manager does, exactly. That is, if I pass in the command registry, it could call any command that JupyterLab has, sometimes in opaque or confusing ways.

Overall, this is not a huge deal, it's largely me being finicky about APIs.

Reply:
No that makes sense. It'd be nice to pass just the one command down to the cluster manager if there needs to be a top level registry. Would you prefer that I use the more react-y approach?

Reply:
I have a slight preference for the react-y approach, but if it winds up being too much of a pain, I am happy to take the `CommandRegistry` approach.

Reply:
I guess since the command exists in the `CommandRegistry` we should use it in the button. So I think I'll leave it as the command-based approach unless you dislike it too much.